### PR TITLE
Feature | BSSH V2 | Adding appsessions support | CROMSO-261 

### PR DESCRIPTION
**Pull request by Shani-Daniel**
[CROMSO-261](https://jira.illumina.com/browse/CROMSO-261)

### CODE REVIEW

**Review by a-lahat — APPROVED** [review]

**Review by a-lahat — DISMISSED** [review]

File: fs_basespace/basespace_context.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from fs import errors
 from fs_basespace.api_factory import BasespaceApiFactory
 
+from BaseSpacePy.model.V2DatasetCompactList import V2DatasetCompactList
```
Comment (yairfischer) [inline fs_basespace/basespace_context.py:8]: remove - it is not used

File: fs_basespace/basespace_context.py [hunk 2]
```diff
@@ -224,9 +236,25 @@ def get_raw_entity_direct(cls, api: BasespaceApiFactory, biosample_id: str, page
                                                 inputbiosamples=biosample_id)
 
 
+class AppSessionsContext(CategoryContextDirect):
+    NAME = "appsessions"
+    ENTITY_CONTEXT = AppSessionContext
+
+    def list_raw(self, api: BasespaceApiFactory, page: Page):
+        offset, limit = translate_page_to_offset_and_limit(page)
+        params = {'sortby': 'Name', 'offset': offset, 'limit': limit, 'output_projects': [self.raw_obj.Id]}
```
Comment (yairfischer) [inline fs_basespace/basespace_context.py:245]: use the get_id method

**Review by yairfischer — CHANGES_REQUESTED** [review]

File: fs_basespace/basespace_context.py (hunk 2 above)
Comment (Shani-Daniel) [reply fs_basespace/basespace_context.py:245]: The method get_id was added in EntityContext, while this object is an instance of CategoryContext

**Review by yairfischer — APPROVED** [review]

File: fs_basespace/basespace_context.py (hunk 2 above)
Comment (yairfischer) [reply fs_basespace/basespace_context.py:245]: ok. correct

**Review by a-lahat — APPROVED** [review]

**Pull request merged**